File: src/leetcode/2026/06/Solution1018.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution1018 {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        vector<bool> result;
        int currentNum = 0;
        for (int num : nums) {
            currentNum = (currentNum * 2 + num) % 5;
            result.push_back(currentNum == 0);
        }
        return result;
    }
};

int main() {
    Solution1018 solution;

    vector<int> nums1 = {0, 1, 1};
    vector<int> nums2 = {1, 1, 1};
    vector<int> nums3 = {0, 1, 1, 1, 1, 1};
    vector<bool> result1 = solution.prefixesDivBy5(nums1);
    vector<bool> result2 = solution.prefixesDivBy5(nums2);
    vector<bool> result3 = solution.prefixesDivBy5(nums3);

    for (bool b : result1) {
        cout << b << " "; // true false false
    }
    cout << endl;

    for (bool b : result2) {
        cout << b << " "; // false false false
    }
    cout << endl;

    for (bool b : result3) {
        cout << b << " "; // true false false false true false
    }
    cout << endl;

    return 0;
}

File: src/leetcode/2026/06/Solution1018.java
import java.util.List;
import java.util.ArrayList;

public class Solution1018 {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        List<Boolean> result = new ArrayList<>();
        int prefixSum = 0;

        for (int num : nums) {
            prefixSum = (prefixSum * 2 + num) % 5;
            result.add(prefixSum == 0);
        }

        return result;
    }

    public static void main(String[] args) {
        Solution1018 solution = new Solution1018();

        int[] nums1 = { 0, 1, 1 };
        System.out.println(solution.prefixesDivBy5(nums1)); // Output: [true, false, false]

        int[] nums2 = { 1, 1, 1 };
        System.out.println(solution.prefixesDivBy5(nums2)); // Output: [false, false, false]

        int[] nums3 = { 0, 1, 1, 1, 0 };
        System.out.println(solution.prefixesDivBy5(nums3)); // Output: [true, false, false, false, false]
    }
}
